### ISSUE TITLE
Working, but one question: Mirabella Genio LED white fairy lights (Australia) 

### ISSUE THREAD

**Issue by Letterman17** [iRayanKhan/homebridge-tuya#284]
I managed to get a set of 500 LED white fairy lights working (Mirabella Genio, Australian brand) with the following:

name": "Genio LED Fairy Lights White",
            "type": "TWLight",
            "manufacturer": "Mirabella",
            "model": "Smart White Bulb",
            "id": "",
            "key": "",
            "dpPower": 1,
            "dpBrightness": 3,
            "dpColorTemperature": 104,
            "minWhiteColor": 25,
            "maxWhiteColor": 255

This is based on the following from the Homebridge log: 
[TuyaLan] Ready to handle Genio LED Fairy Lights White (TWLight:3.3) with signature {"1":true,"3":132,"104":"8"}

So "1" is dpPower, 3 is brightness, but not sure about "104"? There's about 8 steps to adjust the brightness, so is 104:8 somehow related to brightness? Being fairy lights I don't think you can adjust the color temperature actually, only the brightness.

Is there a way to discover the full schema and config for any Tuya device using the Tuya developer account or some other tool to help with Homebridge config? The fairy lights are register in the developer account, but all I seem to see there is:

switch_led | Boolean | {true,false}

bright_value | Integer | {"min":25,"scale":0,"unit":"","max":255,"step":1}

Thanks for helping me understand how this all works.

**Comment by ElphaX** [comment]
So the signature we get back includes all the endpoints that has been exposed to our API usage, usually this is similar to the Tuya Smart or SmartLife apps.

Unfortunately that's all we have to work with currently :(

**Comment by ElphaX** [comment]
> So "1" is dpPower, 3 is brightness, but not sure about "104"? There's about 8 steps to adjust the brightness, so is 104:8 somehow related to brightness? Being fairy lights I don't think you can adjust the color temperature actually, only the brightness.

Yes it would seem that the 104 corresponds to the equivalent brightness interval

> Is there a way to discover the full schema and config for any Tuya device using the Tuya developer account or some other tool to help with Homebridge config? The fairy lights are register in the developer account, but all I seem to see there is:

So the signature we get back includes all the endpoints that has been exposed to our API usage, usually this is similar to the Tuya Smart or SmartLife apps.

Unfortunately that's all we have to work with currently :(
